### PR TITLE
fix(slider): handle right click on mouse down

### LINKED ISSUE #576: Slider remains in pointerDown mode after dismissing right click
### Current Behavior

If you right click on the slider handle and then release or dismiss the right click menu, the slider remains in pointer down mode and moves with the mouse movement.

### Expected behavior

The slider handle should not follow the thumb.

### Reproducible example

This screenshot was taken using Chrome 81 on macOS Catalina at https://reacttraining.com/reach-ui/slider/

Screen-Recording-2020-05-07-at-1

### Suggested solution(s)

The `pointerDownRef.current` ref in the `Slider` component should reset to false when the right click is released.

### Additional context

N/A

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Reach Package    | @reach/slider | 0.10.1 |
| React            |  react | 16.13.1  |
| Browser          | Chrome |  81.0.4044.129 |
| Operating System | MacOS Catalina | 10.15.4 |

### PR DESCRIPTION
**Pull request by raunofreiberg**
Thank you for contributing to Reach UI! Please fill in this template before submitting your PR to help us process your request more quickly.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code (Compile and run).
- [ ] Add or edit tests to reflect the change (Run with `yarn test`).
- [ ] Add or edit Storybook examples to reflect the change (Run with `yarn start`).
- [x] Ensure formatting is consistent with the project's Prettier configuration.

This pull request:

- [ ] Creates a new package
- [x] Fixes a bug in an existing package
- [ ] Adds additional features/functionality to an existing package
- [ ] Updates documentation or example code
- [ ] Other

<hr />

Closes #576

### CODE REVIEW

File: packages/slider/src/index.tsx [hunk 1]
```diff
@@ -513,6 +514,8 @@ const SliderInput = forwardRefWithAs<
   }, [onMouseMove, onMouseDown, onMouseUp, onTouchStart, onTouchEnd, onTouchMove, onPointerDown, onPointerUp]);
 
   let handleSlideStart = useEventCallback((event: SomePointerEvent) => {
+    if (isRightClick(event as MouseEvent)) return;
```
[the remark this answers is not in the document]
Comment (chancestrickland) [reply packages/slider/src/index.tsx:517]: We may also need to handle the pointer event here. Not 100% sure but it is probably worth testing to make sure we don't miss any edge cases there.

[the remark this answers is not in the document]
Comment (raunofreiberg) [reply packages/slider/src/index.tsx:517]: Do you mean pointer event like a pen/stylus?

[the remark this answers is not in the document]
Comment (chancestrickland) [reply packages/slider/src/index.tsx:517]: We are using `onPointerDown` and `onPointerUp` to leverage the `setPointerCapture` API which allows set and release to work when events occur outside of the frame or document. Those will both fire when `mousedown`/`touchstart` and `mouseup`/`touchend` events occur. I haven't yet checked to see if the API for detecting right clicks in a mouse event interface will work on a pointer event object, but we probably want to check on that and prevent `setPointerCapture` from being called in that case.

[the remark this answers is not in the document]
Comment (raunofreiberg) [reply packages/slider/src/index.tsx:517]: Detecting a right click with pointer events seem to work the same as mouse events. [Here's a small demo.](https://codesandbox.io/s/compassionate-mclaren-lhkj8?file=/src/App.js)

[the remark this answers is not in the document]
Comment (raunofreiberg) [reply packages/slider/src/index.tsx:517]: To clarify –– what you're saying is that we should add the same prevention here?

https://github.com/reach/reach-ui/blob/9e4b9adab7fd3085b1fb4b54882a2231308b7416/packages/slider/src/index.tsx#L543

**Comment by chancestrickland** [comment]
Can you please open a new PR to the `develop` branch? Thanks!

**Pull request closed without merging**